src/scribe_data/check/check_data_type_metadata.py

```diff
@@ -0,0 +1,54 @@
+from scribe_data.cli.cli_utils import (
+    LANGUAGE_DATA_EXTRACTION_DIR,
+    data_type_metadata,
+)
+
+
+def check_data_type_metadata(output_file):
+    """
+    Check that subdirectories named for data types in language directories
+    are also reflected in the data_type_metadata.json file, accounting for meta-languages.
+    """
+    # Extract valid data types from data_type_metadata
+    valid_data_types = set(data_type_metadata.keys())
+
+    def check_language_subdirs(lang_dir, meta_lang=None):
+        discrepancies = []
+        
+        for language in lang_dir.iterdir():
+            if language.is_dir():
+                meta_language = meta_lang or language.name.lower()
+                data_types_in_dir = []
+                
+                for data_type in language.iterdir():
+                    if data_type.is_dir():
+                        data_types_in_dir.append(data_type.name.lower())
+                
+                # Compare with valid data types
+                missing_data_types = set(data_types_in_dir) - valid_data_types
+                extra_data_types = valid_data_types - set(data_types_in_dir)
+                
+                if missing_data_types:
+                    discrepancies.append(f"Missing in metadata for '{meta_language}': {missing_data_types}")
+                if extra_data_types:
+                    discrepancies.append(f"Extra in directory for '{meta_language}': {extra_data_types}")
+                
+                # Recursively check sub-languages (if applicable)
+                sub_lang_dir = language / 'sub-languages'
+                if sub_lang_dir.exists():
+                    discrepancies.extend(check_language_subdirs(sub_lang_dir, meta_language))
+        
```

Comment:
hmm...🤔

Reply:
I see...`sub_lang_dir = language / 'sub-languages'` is written to even support the new flow coming from #402, yeah? @catreedle

Reply:
Yeah, it was done keeping the generalisation in mind

Reply:
I dont get it. does this mean the directory structure will change?
Norwegian
--sub-languages
----Nynorks
----Bokmal

Reply:
> I dont get it. does this mean the directory structure will change? Norwegian --sub-languages ----Nynorks ----Bokmal

No, @catreedle, it will remain as is.

Reply:
> does this mean the directory structure will change?

No, the directory structure will remain same, but the scripts will be able to check the data types under sub languages also